pkg/crc/segment/segment.go

```diff
@@ -71,8 +73,14 @@ func (c *Client) Upload(action string, duration time.Duration, err error) error
 		return err
 	}
 
-	properties := analytics.NewProperties().
-		Set("version", version.GetCRCVersion()).
+	properties := analytics.NewProperties()
+	contextProperties := telemetry.PropertiesFromContext(ctx)
+	if contextProperties != nil {
```

Comment:
@guillaumerose do you think with current implementation `contextProperties` is going to `nil`? In the root we use `ContextWithEmptyProperties` where we already do `return context.WithValue(ctx, key, &Properties{storage: make(map[string]interface{})})` which means is is already assigned and not going to be nil, or this is something from the daemon side?

Reply:
With the refactoring Christophe suggested, it's better and it avoids nil checks. PTAL :)